src/reyna/Power/PowerManager.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace Reyna.Power
+{
+    public class PowerManager
+    {
+        public const byte Online = 0x01;
+
+        public SystemPowerStatus SystemPowerStatus
+        {
+            get
+            {
+                var systemPowerStatus = new SystemPowerStatus();
+                if (NativeMethods.GetSystemPowerStatusEx(systemPowerStatus, 1) == 1)
+                {
+                    return systemPowerStatus;
+                }
+
+                return null;
+            }
+        }
+
+        internal bool IsBatteryCharging()
+        {
+            if (this.SystemPowerStatus != null)
```

Comment:
you need to cache the SystemPowerStatus as this will make two calls to the API.
@Youhana-Hana